index.js

```diff
@@ -87,41 +118,43 @@
             return dom.toDataURL(type, quality);
         },
 
-        isDataURL: function(str) {
-            regex = /^\s*data:([a-z]+\/[a-z]+(;[a-z\-]+\=[a-z\-]+)?)?(;base64)?,[a-z0-9\!\$\&\'\,\(\)\*\+\,\;\=\-\.\_\~\:\@\/\?\%\s]*\s*$/i;
+        isDataURL(str) {
+            var regex = /^\s*data:([a-z]+\/[a-z]+(;[a-z\-]+\=[a-z\-]+)?)?(;base64)?,[a-z0-9\!\$\&\'\,\(\)\*\+\,\;\=\-\.\_\~\:\@\/\?\%\s]*\s*$/i;
             return !!str.match(regex);
         },
 
-        loadImage: function (imageURL) {
+        loadImage(imageURL) {
             var imageObj = new Image();
             imageObj.onload = this.handleImageReady.bind(this, imageObj);
             if (!this.isDataURL(imageURL)) imageObj.crossOrigin = 'anonymous';
             imageObj.src = imageURL;
         },
 
-        componentDidMount: function () {
+        componentDidMount() {
             var context = this.getDOMNode().getContext('2d');
             if (this.props.image) {
                 this.loadImage(this.props.image);
             }
             this.paint(context);
-            document && document.addEventListener('mousemove', this.handleMouseMove, false);
-            document && document.addEventListener('mouseup', this.handleMouseUp, false);
+            document.addEventListener(deviceEvents.native.move, this.handleMouseMove, false);
+            document.addEventListener(deviceEvents.native.up, this.handleMouseUp, false);
+
+            if (isTouchDevice) React.initializeTouchEvents(true);
         },
 
-        componentWillUnmount: function () {
-            document && document.removeEventListener('mousemove', this.handleMouseMove, false);
-            document && document.removeEventListener('mouseup', this.handleMouseUp, false);
+        componentWillUnmount() {
+            document.removeEventListener(deviceEvents.native.move, this.handleMouseMove, false);
+            document.removeEventListener(deviceEvents.native.up, this.handleMouseUp, false);
```

Comment:
The missing "document" was causing problems in nodejs environment. Removed this on purpose?